admin/featured_categories.php

```diff
@@ -0,0 +1,550 @@
+
+        zen_redirect(zen_href_link(FILENAME_FEATURED_CATEGORIES, ($currentPage != 0 ? 'page=' . $currentPage . '&' : '') . (isset($_GET['search']) ? '&search=' . $_GET['search'] . '&' : '') . (isset($new_featured) ? 'fID=' . $new_featured->fields['featured_categories_id'] : '')));
+      }
+      break;
+    case 'update':
+        $featured_categories_id = (int)$_POST['featured_categories_id'];
+        $error = false;
+
+        $featured_date_available_raw = zen_db_prepare_input($_POST['featured_date_available']);
+        if ($featured_date_available_raw === '') {
+            $featured_date_available = '0001-01-01';
+        } else {
+            if (DATE_FORMAT_DATE_PICKER !== 'yy-mm-dd' && !empty($featured_date_available_raw)) {
+                $local_fmt = zen_datepicker_format_fordate();
+                $dt = DateTime::createFromFormat($local_fmt, $featured_date_available_raw);
+                $featured_date_available_raw = 'null';
+                if (!empty($dt)) {
+                    $featured_date_available_raw = $dt->format('Y-m-d');
+                }
+            }
+            if (zcDate::validateDate($featured_date_available_raw) === true) {
+                $featured_date_available = $featured_date_available_raw;
+            } else {
+                $error = true;
+                $messageStack->add(ERROR_INVALID_AVAILABLE_DATE, 'error');
+            }
+        }
+
+        $expires_date_raw = zen_db_prepare_input($_POST['expires_date']);
+        if ($expires_date_raw === '') {
+            $expires_date = '0001-01-01';
+        } else {
+            if (DATE_FORMAT_DATE_PICKER !== 'yy-mm-dd' && !empty($expires_date_raw)) {
+                $local_fmt = zen_datepicker_format_fordate();
+                $dt = DateTime::createFromFormat($local_fmt, $expires_date_raw);
+                $expires_date_raw = 'null';
+                if (!empty($dt)) {
+                    $expires_date_raw = $dt->format('Y-m-d');
+                }
+            }
+            if (zcDate::validateDate($expires_date_raw) === true) {
+                $expires_date = $expires_date_raw;
+            } else {
+                $error = true;
+                $messageStack->add(ERROR_INVALID_EXPIRES_DATE, 'error');
+            }
+        }
+
+        if ($error === true) {
+            $action = 'edit';
+            break;
+        }
+
+        $db->Execute("UPDATE " . TABLE_FEATURED_CATEGORIES . "
+                    SET featured_last_modified = now(),
+                        expires_date = '" . zen_db_input($expires_date) . "',
+                        featured_date_available = '" . zen_db_input($featured_date_available) . "'
+                    WHERE featured_categories_id = " . (int)$featured_categories_id);
+
+        zen_redirect(zen_href_link(FILENAME_FEATURED_CATEGORIES, ($currentPage != 0 ? 'page=' . $currentPage . '&' : '') . (isset($_GET['search']) ? 'search=' . $_GET['search'] . '&' : '') . 'fID=' . (int)$featured_categories_id));
```

Comment:
`(int)` casting of `$featured_categories_id` not needed here; that was already done at line 92.